packages/teams-js/src/public/media.ts

```diff
@@ -739,82 +689,46 @@ export namespace media {
-    } else {
-      if (typeof callbackOrConfig === 'object') {
-        // the first parameter is an object - it's the config! No callback.
-        config = callbackOrConfig;
-      } else {
-        // otherwise, it's a function, so a callback. The second parameter might be a callback
-        callback = callbackOrConfig;
-        config = configMaybe;
-      }
+  export function scanBarCode(callback: (error: SdkError, decodedText: string) => void, config?: BarCodeConfig): void {
+    if (!callback) {
+      throw new Error('[media.scanBarCode] Callback cannot be null');
     }
-
     ensureInitialized(FrameContexts.content, FrameContexts.task);
 
-    const wrappedFunction: InputFunction<string> = () =>
-      new Promise<string>(resolve => {
-        if (
-          GlobalVars.hostClientType === HostClientType.desktop ||
-          GlobalVars.hostClientType === HostClientType.web ||
-          GlobalVars.hostClientType === HostClientType.rigel ||
-          GlobalVars.hostClientType === HostClientType.teamsRoomsWindows ||
-          GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
-          GlobalVars.hostClientType === HostClientType.teamsPhones ||
-          GlobalVars.hostClientType === HostClientType.teamsDisplays
-        ) {
-          throw { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
-        }
-
-        if (!isCurrentSDKVersionAtLeast(scanBarCodeAPIMobileSupportVersion)) {
-          throw { errorCode: ErrorCode.OLD_PLATFORM };
-        }
+    if (
+      GlobalVars.hostClientType === HostClientType.desktop ||
+      GlobalVars.hostClientType === HostClientType.web ||
+      GlobalVars.hostClientType === HostClientType.rigel ||
+      GlobalVars.hostClientType === HostClientType.teamsRoomsWindows ||
+      GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
+      GlobalVars.hostClientType === HostClientType.teamsPhones ||
+      GlobalVars.hostClientType === HostClientType.teamsDisplays
+    ) {
+      const notSupportedError: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
+      callback(notSupportedError, null);
+      return;
+    }
 
-        if (!validateScanBarCodeInput(config)) {
-          throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
-        }
+    if (!isCurrentSDKVersionAtLeast(scanBarCodeAPIMobileSupportVersion)) {
+      const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
+      callback(oldPlatformError, null);
+      return;
+    }
 
-        resolve(sendAndHandleSdkError('media.scanBarCode', config));
-      });
+    if (!validateScanBarCodeInput(config)) {
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      callback(invalidInput, null);
+      return;
+    }
 
-    return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<string>(wrappedFunction, callback);
+    sendMessageToParent('media.scanBarCode', [config], callback);
   }
-
   export function isSupported(): boolean {
```

Comment:
v1 does not have isSupported function, but I think we would want to keep it?

Reply:
No, we won't be adding isSupported infrastructure to namespaces that haven't been assessed for v2.0 alignment, so not adding any additional gates to the APIs under this namespace. Can you please remove this?

Reply:
There will be changes necessary in the runtime.ts file as well, since this change should also remove `media` from the `IRuntime` interface.